Metric App Seup

Carshy opened the pull request:
In the current project, I build a mobile web application to check a list of metrics (numeric values). I created this project using React and Redux.

I selected an API that provides numeric data

DammyShittu commented on src/components/Animals/Search.js:
```diff
@@ -0,0 +1,34 @@
+// import React, { useEffect, useState } from 'react';
+// import { useSelector } from 'react-redux';
+
+// const Search = () => {
+//   const animals = useSelector((state) => state.animals);
+//   const [newAnimals, setNewAnimals] = useState([]);
+
+//   useEffect(() => {
+//     setNewAnimals(animals);
+//   }, [animals, setNewAnimals]);
+
+//   const handleSearch = (e) => {
+//     e.preventDefault();
+//     let myValue = e.target.value;
+//     if (myValue.length > 0) {
+//       myValue = myValue.toLowerCase();
+//       const filteredAnmls = animals.filter((animal) =>
+// animal.name.toLowerCase().includes(myValue));
+//       setNewAnimals(filteredAnmls);
+//     } else {
+//       setNewAnimals(animals);
+//     }
+//   };
+
+//   return (
+//     <div>
+//       <div className="searchbar">
+//         <input type="text" placeholder="Search eg Lion..." onChange={handleSearch} />
+//       </div>
+//     </div>
+//   );
+// };
+
+// export default Search;
```
- [ ] Kindly remove this file of commented code if you would not be using it, but get rid of the commented code. It makes your code look messy

BenMukebo replied:
> * [ ]  Kindly remove this file of commented code if you would not be using it, but get rid of the commented code. It makes your code look messy

- [ ] As requested by the previous reviewer, please remove do not leave unused code as a comment in your codebase. this will make your code much more noisy, It is always a good practice to keep your code clean from such comments. 😉 👍🏻

DammyShittu commented on src/components/Animals/Animals.js:
```diff
@@ -0,0 +1,48 @@
+import { fetchAnimalsDataFromApi } from '../../redux/animalia/animalia';
+
+function Animals() {
+// get Animals data from the store
+  const animals = useSelector((state) => state.animals);
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    if (!animals.length) {
+      dispatch(fetchAnimalsDataFromApi());
+    }
+  }, [animals.length, dispatch]);
+  return (
+    <div>
+      <Dropdown />
+      <div className="animal-wrapper">
+        {
+          animals.map((animal) => (
+            <Animal
+              key={animal.id}
+              id={animal.id}
+              image={animal.image_link}
+              name={animal.name}
+              type={animal.animal_type}
+              latinName={animal.latin_name}
+              activeTime={animal.active_time}
+              minLength={animal.length_min}
+              maxLength={animal.length_max}
+              minWeight={animal.weight_min}
+              maxWeight={animal.weight_max}
+              lifeSpan={animal.lifespan}
+              habitats={animal.habitat}
+              diet={animal.diet}
+              locations={animal.geo_range}
+            />
+          ))
+        }
+      </div>
+    </div>
+  );
```
- [ ] Your home page should contain a search functionality for filtering items on the home page. Kindly implement this

  ![gim](https://user-images.githubusercontent.com/54848363/189332404-dcdcd43e-dd5c-4a77-afe8-b081a9882b6d.jpg)

BenMukebo replied:
> * [ ]  Your home page should contain a search functionality for filtering items on the home page. Kindly implement this
>   ![gim](https://user-images.githubusercontent.com/54848363/189332404-dcdcd43e-dd5c-4a77-afe8-b081a9882b6d.jpg)

- [ ] As requested by the previous reviewer, please make sure the Home page has a search functionality for filtering your animal list using given parameters, e.g. animal. name.

DammyShittu reviewed (changes requested):
Hi @Carshy,

## Highlights
✔️ Descriptive PR
✔️ Passing linters
✔️ Good branch name

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
- [ ] Your readme documentation should contain acknowledgment for Nelson Sakwa whose design was used for the project.

- [ ] Ensure your deploy your application and provide the link in your readme doc

- [ ] Kindly record your video presentation and include the link in the readme doc

- [ ] You currently have no tests. Kindly write unit and integration tests for your functions and components respectively.

- [ ] There seems to be something wrong with your screenshots. Kindly fix that to maintain a good documentation

  ![svv](https://user-images.githubusercontent.com/54848363/189333463-0951e4e4-70b6-444f-bb0a-76da7ebb9c0a.jpg)

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

Carshy commented:
Hello, I have implemented all changes as requested by the previous reviewer: Highlighted changes, ReadMe acknowledgement to Nelson Sakwa from Behance, Implemented testing, inserted screenshots, Recorded a video and inserted link in README, deployed my application. @DammyShittu thanks for the review it was really helpful

BenMukebo commented on src/App.js:
```diff
@@ -1,12 +1,25 @@
+import Animals from './components/Animals/Animals';
+import AnimalDetails from './components/Animals/AnimalDetails';
+import Header from './components/Header/Header';
+// import Dropdown from './components/Animals/Dropdown';
```
- [ ] Same here, Please do not leave unused code as a comment in your codebase. this will make your code much more noisy, It is always a good practice to keep your code clean from such comments. 😉 👍🏻

BenMukebo reviewed (changes requested):
## Hi @Carshy, 

Great work on making the changes requested by a previous reviewer 👏 
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>


### To Highlight 🎉
- You have followed  JavaScript best practices. ✔️ 
- Professional README file. :heavy_check_mark:
- PR has a good title and summary. ✔️
- Good presentation video. ✔️ 
- Your app has been well deployed using to GitHub Pages. ✔️ 
- Your Design looks awesome, keep rocking :rocket: 

## STATUS: CHANGES REQUESTED  ♻️ 
_Please Check the comments under the review and make all required changes_
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers :clinking_glasses:  and Happy coding!!! :dancers:💻  

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please ping me @BenMukebo when you comment so I can receive the notification.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

Carshy commented:
Changes made as requested. @BenMukebo thanks for your review it was really helpful

emmanuellekamwa reviewed (approved):
**STATUS: APPROVED 🍾✔🎉**
Hello @Carshy 👋
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

- You documented your work professionally.✔
- Your linters are passing all the checks
- You asked for the changes in a polite and friendly way.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
